src/test/java/io/confluent/connect/elasticsearch/integration/ElasticSearchSinkConnectorIT.java

```diff
@@ -0,0 +1,315 @@
+        KAFKA_TOPIC);
+    log.info("Number of records added in kafka {}", totalRecords.count());
+
+    // Configure Connector and wait some specific time to start the connector.
+    connect.configureConnector(CONNECTOR_NAME, props);
+    waitForConnectorToStart(CONNECTOR_NAME, Integer.valueOf(MAX_TASKS));
+
+    // Wait Connector to write data into elastic-search
+    waitForConnectorToWriteDataIntoES(
+        CONNECTOR_NAME,
+        Integer.valueOf(MAX_TASKS),
+        KAFKA_TOPIC,
+        NUM_RECORDS);
+    assertRecordsCountAndContent(NUM_RECORDS, KAFKA_TOPIC);
+    pumbaPauseContainer.close();
+  }
+
+  @Test
+  public void testForElasticSearchServerDelay() throws InterruptedException, IOException {
+    startPumbaDelayContainer();
+    sendTestDataToKafka(0, NUM_RECORDS);
+    ConsumerRecords<byte[], byte[]> totalRecords = connect.kafka().consume(
+        NUM_RECORDS,
+        CONSUME_MAX_DURATION_MS,
+        KAFKA_TOPIC);
+    log.info("Number of records added in kafka {}", totalRecords.count());
+
+    // Configure Connector and wait some specific time to start the connector.
+    connect.configureConnector(CONNECTOR_NAME, props);
+    waitForConnectorToStart(CONNECTOR_NAME, Integer.valueOf(MAX_TASKS));
+
+    // Wait Connector to write data into elastic-search
+    waitForConnectorToWriteDataIntoES(
+        CONNECTOR_NAME,
+        Integer.valueOf(MAX_TASKS),
+        KAFKA_TOPIC,
+        NUM_RECORDS);
+    assertRecordsCountAndContent(NUM_RECORDS, KAFKA_TOPIC);
+    pumbaDelayContainer.close();
+  }
+
+  @Test
+  public void testCredentialChange() throws InterruptedException, IOException {
+    sendTestDataToKafka(0, NUM_RECORDS);
+    ConsumerRecords<byte[], byte[]> totalRecords = connect.kafka().consume(
+        NUM_RECORDS,
+        CONSUME_MAX_DURATION_MS,
+        KAFKA_TOPIC);
+    log.info("Number of records added in kafka {}", totalRecords.count());
+
+    // Configure Connector and wait some specific time to start the connector.
+    connect.configureConnector(CONNECTOR_NAME, props);
+    waitForConnectorToStart(CONNECTOR_NAME, Integer.valueOf(MAX_TASKS));
+
+    // Wait Connector to write data into elastic-search
+    waitForConnectorToWriteDataIntoES(
+        CONNECTOR_NAME,
+        Integer.valueOf(MAX_TASKS),
+        KAFKA_TOPIC,
+        NUM_RECORDS);
+    assertRecordsCountAndContent(NUM_RECORDS, KAFKA_TOPIC);
+
+    int status = changeESpassword("elastic", "elastic1");
+    Assert.assertEquals(200, status);
+
+    sendTestDataToKafka(NUM_RECORDS, NUM_RECORDS);
+    totalRecords = connect.kafka().consume(
+        NUM_RECORDS * 2,
+        CONSUME_MAX_DURATION_MS,
+        KAFKA_TOPIC);
+    log.info("Number of records added in kafka {}", totalRecords.count());
+    // Refresh ES client with new credential at test level to perform search operation
+    refreshESClient();
+    // Second batch of records should not be entertained due to credential change
+    waitForConnectorToWriteDataIntoES(
```

Comment:
This is for asserting the initially inserted records. It won't guarantee you that second batch is not inserted. Either you can remove this part or alter it to check 2nd batch of records which are inserted ko kafka after changing ES password is not present in Elastic Search index.